DOC.md

```diff
@@ -378,6 +379,96 @@ s("trig", {
 If the function only performs simple operations on text, consider using
 the `lambda` from [`luasnip.extras`](#extras)
 
+# POSTFIX SNIPPET
+
+Postfix snippets, famously used in 
+[rust analyzer](https://rust-analyzer.github.io/) and various IDEs, are a type
+of snippet, which alters text before the snippets trigger. While these
+can be implemented using regTrig snippets, this helper makes the process easier
+in most cases.
+
+The simplest example, which surrounds the text preceeding the `.br` with
+brackets `[]`, looks like:
+
+```lua
+
+      postfix(".br", {
+              f(function(_, parent)
+                    return "[" .. parent.snippet.env.POSTFIX_MATCH .. "]"
+              end, {}),
+      })
+```
+
+and is triggered with `xxx.br` and expands to `[xxx]`.
+
+Note the `parent.snippet.env.POSTFIX_MATCH` in the function node. This is additional
+field generated by the postfix snippet. This field is generated by extracting
+the text matched (using a configurable matching string, see below) from before
+the trigger. In the case above, the field would equal `"xxx"`. This is also
+usable within dynamic nodes.
+
+This field can also be used within lambdas and dynamic nodes.
+
+```lua
+      postfix(".br", {
+              l("[" .. l.POSTFIX_MATCH .. "]"),
+      })
+```
+
+```lua
+	postfix(".brd", {
+	  d(1, function (_, parent)
+	    return sn(nil, {t("[" .. parent.env.POSTFIX_MATCH .. "]")})
+	  end)
+	}),
+```
+
+The arguments to `postfix` are identical to the arguments to `s` but with a few
+extra options.
+
+The first argument can be either a string or a table. If it is a string, that
+string will act as the trigger, and if it is a table it has the same valid keys
+as the table in the same position for `s` except:
+
+- `wordTrig`: This key will be ignored if passed in, as it must always be
+    false for postfix snippets.
+- `match_pattern`: The pattern that the line before the trigger is matched
+    against. The default match pattern is `"[%w%.%_%-]+$"`. Note the `$`. This
+    matches since only the line _up until_ the beginning of the trigger is
+    matched against the pattern, which makes the character immediately
+    preceeding the trigger match as the end of the string.
+
+Some other match strings, including the default, are available from the postfix
+module. `require("luasnip.extras.postfix).matches`:
+
+- `default`: [%w%.%_%-%"%']+$
+- 'line': `^.+$`
```

Comment:
I think "line" and the default-pattern are also supposed to be inside backticks, right?

Reply:
yup!

Reply:
i was also thinking of taking out the `"` and `'` in the default pattern, since it wouldn't match on something like `"a string with spaces in it"`

Reply:
`[%w%.%_%-%"%']` would match on `"somestring"`, wouldn't it? That's alright I think

Oh an interesting addition here would be treesitter-awareness, to make the postfix capture the node before the trigger. No pressure to implement this though, I just wanted to throw the thought out there

Reply:
Yeah I've been playing with that idea too. I think in another pr I might give it a shot, but I'd like to see how people use/ need from the normal postfix snips first I think